fedimint-core/src/api.rs

```diff
@@ -595,6 +599,40 @@ impl WsClientConnectInfo {
         }
     }
 
+    /// Converts the connect info into a bech32 string for compact display / copying
+    pub fn to_bech32(&self) -> anyhow::Result<String> {
+        let mut data = vec![];
+        data.extend(self.id.0.to_bytes());
+        for url in &self.urls {
+            data.extend(";".as_bytes());
+            data.extend(url.as_str().as_bytes());
+        }
+
+        Ok(bech32::encode("fed", data.to_base32(), Bech32m)?)
+    }
+
+    /// Parses the connect info from a bech32 string
+    pub fn from_bech32(encoded: &str) -> anyhow::Result<Self> {
+        let (hrp, data, variant) = bech32::decode(encoded)?;
+
+        ensure!(hrp == "fed", "Invalid HRP in bech32 encoding");
+        ensure!(variant == Bech32m, "Expected Bech32m encoding");
+
+        let bytes: Vec<u8> = Vec::<u8>::from_base32(&data)?;
+        let id_bytes: [u8; PK_SIZE] = bytes[..PK_SIZE].try_into()?;
+        let url_str = std::str::from_utf8(&bytes[PK_SIZE + 1..])?;
+        let mut urls = vec![];
+
+        for url in url_str.split(';') {
+            urls.push(url.parse()?);
+        }
+
+        Ok(Self {
+            urls,
+            id: FederationId(PublicKey::from_bytes(id_bytes)?),
+        })
+    }
+
```

Comment:
If this becomes the canonical user-facing representation it should probably rather be implemented as `Display` and `FromStr`.